src/whitenoise/users.rs

```diff
@@ -334,6 +446,106 @@ impl User {
 
         Ok(users_with_relays)
     }
+
+    /// Determines whether metadata should be updated based on comprehensive event processing logic.
+    ///
+    /// This method implements the complete logic for deciding whether to process a metadata event:
+    /// 1. Check if we've already processed this specific event (avoid double processing)
+    /// 2. If user is newly created, always accept the event
+    /// 3. If user has default metadata, always accept the event
+    /// 4. Otherwise, check if the event timestamp is newer than or equal to stored timestamp
+    ///
+    /// # Arguments
+    /// * `event_id` - The ID of the metadata event being considered
+    /// * `event_datetime` - The datetime of the metadata event being considered
+    /// * `newly_created` - Whether this user was just created
+    /// * `database` - Database connection for checking processed events
+    ///
+    /// # Returns
+    /// * `Ok(true)` if metadata should be updated
+    /// * `Ok(false)` if the event should be ignored
+    pub(crate) async fn should_update_metadata(
```

Comment:
Before, we were also checking whether the metadata actually changed, now we don't. I don't think it makes much of a difference other than it can potentially save us a couple of queries